JWBestPracticeApps/AdvancedPlayer/AdvancedPlayer/ViewController.swift

```diff
@@ -0,0 +1,299 @@
+//
+//  ViewController.swift
+//  AdvancedPlayer
+//
+//  Created by Michael Salvador on 1/20/22.
+//
+
+import UIKit
+import JWPlayerKit
+import AVFoundation
+
+// This best practices app demonstrates the following functionality (search for instances of text below to find relevant code)
+// 1. Hide & disable built-in controls/ad controls
+// 2. Add custom controls
+// 3. Observer for 'didFinishLoadingWithTime' event. 'didFinishLoadingWithTime' is analogous to the `onFirstFrame` event in version 3.x.
+// 4. Detect whether an upcoming ad is skippable
+// 5. Time observation for content
+// 6. Time observation for ads
+
+class ViewController: JWPlayerViewController,
+                      JWPlayerViewControllerDelegate,
+                      CustomControlsDelegate {
+
+    private let videoUrlString = "https://playertest.longtailvideo.com/adaptive/oceans/oceans.m3u8"
+    private let posterUrlString = "https://d3el35u4qe4frz.cloudfront.net/bkaovAYt-480.jpg"
+
+    // custom UI
+    var customControls: CustomControls!
+    var progressBarScrubbing: Bool = false
+    var skipOffset: Double?
+    var fullScreenVC: JWFullScreenViewController?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        delegate = self
+
+        // Set the view's background color to black for better contrast.
+        view.backgroundColor = .black
+
+        // MARK: - 1. Hide & disable built-in controls/ad controls
```

Comment:
This is a strange use of `// MARK: -`, since it is typically used for navigating the file from the method/file drop down. Perhaps only have the `1.`?